stackit/internal/resources/kubernetes/validations.go

```diff
@@ -60,6 +62,21 @@ func (r Resource) validate(
 	return nil
 }
 
+// maxVersionOption returns the maximal version that matches the given version.
+// If the given version only contains major and minor version, the latest patch version is returned.
+func maxVersionOption(version string, versionOptions []options.KubernetesVersion) string {
+	if strings.Count(version, ".") == 2 {
+		return version
+	}
+	ret := version
+	for _, v := range versionOptions {
+		if len(v.Version) > len(version) && strings.HasPrefix(v.Version, version+".") && v.Version > ret {
```

Comment:
I'm wondering if we should use something like [semver.Compare()](https://pkg.go.dev/golang.org/x/mod/semver#Compare)

it seems that if I have a version 1.23 and options:
1.23.3
1.23.2
1.23.1

your code will probably choose 1.23.1 instead of 1.23.3